src/citrine/resources/project.py

```diff
@@ -237,14 +238,44 @@ def ara_definitions(self) -> TableConfigCollection:  # pragma: no cover
              "Please call table_configs instead.", DeprecationWarning)
         return self.table_configs
 
-    def share(self,
-              project_id: str,
-              resource_type: str,
-              resource_id: str) -> Dict[str, str]:
-        """Share a resource with another project."""
+    def share(self, *,
+              resource: Optional[Resource] = None,
+              project_id: Optional[Union[str, UUID]] = None,
+              resource_type: Optional[str] = None,
+              resource_id: Optional[str] = None
+              ) -> Dict[str, str]:
+        """Share a resource with another project.
+
+        Parameters
+        ----------
+        resource: Resource
+            The resource owned by this project, which will be shared
+        project_id: Union[str, UUID]
+            The id of the project with which to share the resource
+        resource_type: Optional[str]
+            [DEPRECATED] Please use ``resource`` instead
+            The type of the resource to share. Must be one of DATASET, MODULE, USER,
+            TABLE, OR TABLE_DEFINITION
+        resource_id: Optional[str]
+            [DEPRECATED] Please use ``resource`` instead
+            The id of the resource to share
+
+        """
+        resource_dict = None
+        if resource is not None:
+            resource_dict = resource.as_entity_dict()
```

Comment:
I think this `as_entity_dict` name is a bit opaque: I had to look up what that method did.  Could you add a comment here, or change the function name to something more self-documenting?

Reply:
I'll add a comment. Note that it's the same method used by `transfer_resource`, `make_public`, and `make_private`

Reply:
Decided to change the name instead